test/data/gen_ch/example_test.c.jinja

```diff
@@ -489,6 +489,153 @@ int refda_adm_example_test_init(refda_agent_t *agent)
+                    // ari://example/test/CTRL/first
+                    // FIXME reference to unknown object
+                }
+                {
+                    cace_ari_t *item = cace_ari_list_push_back_new(acinit.items);
+                    // ari://example/test/CTRL/second
+                    // FIXME reference to unknown object
+                }
+                cace_ari_set_ac(&(objdata->action), &acinit);
+            }
+            // condition
+            {
+                cace_ari_ac_t acinit;
+                cace_ari_ac_init(&acinit);
+                {
+                    cace_ari_t *item = cace_ari_list_push_back_new(acinit.items);
+                    // ari://example/test/EDD/sensor
+                    // FIXME reference to unknown object
+                }
+                {
+                    cace_ari_t *item = cace_ari_list_push_back_new(acinit.items);
+                    // ari://example/test/VAR/min_threshold
+                    // FIXME reference to unknown object
+                }
+                {
+                    cace_ari_t *item = cace_ari_list_push_back_new(acinit.items);
+                    // ari://example/test/OPER/compare_lt
+                    // FIXME reference to unknown object
+                }
+                cace_ari_set_ac(&(objdata->condition), &acinit);
+            }
+            // min_interval
+            {
+              struct timespec ts = {0, 0};
+              cace_ari_set_td(&(objdata->min_interval), ts);
+            }
+            // init_enabled
+            objdata->init_enabled = true;
+            // max_exec_count
+            objdata->max_exec_count = 0;
+
+            obj = refda_register_sbr(adm, cace_amm_idseg_ref_withenum("sbr2", REFDA_ADM_EXAMPLE_TEST_ENUM_OBJID_SBR_SBR2), objdata);
+        }
+
+        /**
+         * Register TBR objects
+         */
+        { // For ./TBR/tbr_rule
+            refda_amm_tbr_desc_t *objdata = CACE_MALLOC(sizeof(refda_amm_tbr_desc_t));
+            refda_amm_tbr_desc_init(objdata);
+            // action
+            {
+                cace_ari_ac_t acinit;
+                cace_ari_ac_init(&acinit);
+                {
+                    cace_ari_t *item = cace_ari_list_push_back_new(acinit.items);
+                    // ari://example/test/CTRL/first
+                    // FIXME reference to unknown object
+                }
+                {
+                    cace_ari_t *item = cace_ari_list_push_back_new(acinit.items);
+                    // ari://example/test/CTRL/other
+                    // FIXME reference to unknown object
+                }
+                cace_ari_set_ac(&(objdata->action), &acinit);
+            }
+            // period
+            {
+              struct timespec ts = {30, 0};
+              cace_ari_set_td(&(objdata->period), ts);
+            }
+            // start_time
+            {
+              struct timespec ts = {946684820, 5};
```

Comment:
This is where it doesn't correspond with the input `/TP/20.5`, which would be `{CACE_ARI_DTN_EPOCH + 20, 500000000}`. Using the CACE macro may actually be good to do in the generator, since the ARI includes are guaranteed anyway.

Reply:
Using `CACE_ARI_DTN_EPOCH` is a good idea however ace is already incorporating the DTN epoch. With the above changes to emit proper nanosecond values we are now generating the proper timestamps.